### ISSUE TITLE
Consider forcing remount with some heuristics

### ISSUE THREAD

**Issue by gaearon** [gaearon/react-proxy#15]
If constructor, `componentWillMount`, or `componentDidMount` code changed, we should probably force remounting.

**Comment by Yomguithereal** [comment]
This would be a very comfortable feature indeed. Can I help you with this somehow @gaearon?

**Comment by gaearon** [comment]
Yeah, feel free to experiment. You'd need to modify the code to create a new proxy class and prototype under some conditions and return them from `get()` after “incompatible” `update()`.

**Comment by yonatanmn** [comment]
any update here? @Yomguithereal

**Comment by creeperyang** [comment]
Related to <https://github.com/gaearon/react-proxy/pull/39>. Best to consider things Comprehensively.
